design/proof-network.md

```diff
@@ -0,0 +1,257 @@
+    storage provider  --------------------------------------->  validator
+
+
+The validator will include the transaction in the next block that it proposes to
+the consensus network. When the proposed block is sequenced by the consensus
+network, the validator will return an inclusion proof to the provider. This is a
+proof that the transaction was included in a block, and that the consensus
+network included the block.
+
+
+                                              inclusion proof
+    storage provider  <---------------------------------------  validator
+
+
+This inclusion proof doesn't need to be succinct, and can for example consist of
+a record of the messages that were exchanged between validators as part of the
+consensus protocol.
+
+Notice that validators do not check the correctness of `StorageProof`
+transactions prior to including them in blocks. In this design sequencing and
+evaluation of transactions are separated. First, the validators reach consensus
+on a sequence of blocks of transactions. Then, each of the validators evaluates
+these transactions in order.
+
+When evaluating a `StorageProof` transaction a validator checks that it was
+submitted within the *period* and that the *proof* is correct w.r.t. to the
+*inputs*. If that is all in order then it updates its internal accounting to
+reflect that the proof was submitted and correct.
+
+### Missing proofs ###
+
+Validators monitor the on-chain marketplace to check which slots require a
+storage proof to be submitted and what the public *inputs* for the proof are.
+For each required proof they check at the end of its *period* whether that proof
+was submitted and correct. If they did not receive a correct proof then they
+will add a `MissingProof(slot id, period, inputs)` transaction to the next
+block.
+
+The `MissingProof` transactions are sequenced by the consensus algorithm. They
+are then evaluated by each validator. They will note the first `MissingProof`
+transaction that correctly notices a missing proof, and allow the sender of that
+first transaction to go on-chain to mark the proof as missing.
+
+The validator that sent the first `MissingProof` transaction for a missing proof
+can now request BLS signatures from the other validators to enact on-chain
+slashing of the storage provider for missing a proof. When the validator
+receives enough signatures to represent > 2/3 stake it can combine these
+signatures into a single combined BLS signature. The validator can then submit
+*slot id*, *period*, *inputs* and the combined signature to the marketplace.
+
+The marketplace will then verify the correctness of *inputs* for the *slot id*
+and *period*, and checks that the combined signature is representative for > 2/3
+stake. If these conditions are met, it will then slash the storage provider
+collateral and reward the validator.
+
+### Faulty proofs ###
+
+The storage proofs that a storage provider submits can be faulty for a number of
```

Comment:
Is it possible to assign different penalties to these faulty proofs?

Reply:
i don't think that would make sense, a proof is either present and valid, or not. There are no "qualities" of missing/wrong proofs.

A more interesting case when storage providers also aggregate their proofs. Then it can happen that for example they submit a single proof for 100 slots, but only 99 is valid because they actually lost 1.